modules/misunderstood/src/backend/db.test.ts

```diff
@@ -0,0 +1,218 @@
+import Db, { TABLE_NAME } from './db'
+import { FLAGGED_MESSAGE_STATUS, FLAG_REASON } from '../types'
+import 'reflect-metadata'
+import Database from '../../../../src/bp/core/database'
+
+import { createDatabaseSuite } from '../../../../src/bp/core/database/index.tests'
+
+createDatabaseSuite('Misunderstood - DB', (database: Database) => {
+  const db = new Db({ database: database.knex })
+
+  beforeAll(async () => {
+    db.knex = database.knex
+    await db.initialize()
+    const statement = db.knex.isLite ? 'DELETE FROM' : 'TRUNCATE TABLE'
+    await db.knex.raw(`${statement} "${TABLE_NAME}";`)
+  })
+
+  beforeEach(async () => {
+    await db.knex.raw(db.knex.isLite ? 'BEGIN TRANSACTION' : 'START TRANSACTION')
+  })
+
+  afterEach(async () => {
+    await db.knex.raw(db.knex.isLite ? 'ROLLBACK TRANSACTION' : 'ROLLBACK')
+  })
+
+  describe('listEvents', () => {
+    it('Returns no events when none in table', async () => {
+      const botId = 'mybot'
+      const language = 'en'
+
+      const events = await db.listEvents(botId, language, FLAGGED_MESSAGE_STATUS.new)
+      expect(events).toHaveLength(0)
+    })
+
+    it('Returns one event if one exists', async () => {
+      const botId = 'mybot'
+      const eventId = '1234'
+      const language = 'en'
+      await db.addEvent({ botId, eventId, language, preview: 'some message', reason: FLAG_REASON.action })
+
+      const events = await db.listEvents(botId, language, FLAGGED_MESSAGE_STATUS.new)
+
+      expect(events).toHaveLength(1)
+    })
+
+    it('Returns many events if many exist', async () => {
+      const botId = 'mybot'
+      const eventId = '1234'
+      const language = 'en'
+      for (let i = 0; i < 3; i++) {
+        await db.addEvent({ botId, eventId, language, preview: 'some message', reason: FLAG_REASON.action })
+      }
+
+      const events = await db.listEvents(botId, language, FLAGGED_MESSAGE_STATUS.new)
+
+      expect(events).toHaveLength(3)
+    })
+
+    it('Filters - by botId', async () => {
+      const props = {
```

Comment:
I know this is a nitty but can you extract this as a constant and re-use it in all you tests. If we change the interface then we'll have to change it at a single place to make all the tests pass